File: packages/csharp/src/SheetCompressor/Baseline.cs
using System.Collections.Generic;
using System.Linq;

namespace SheetCompressor;

public static class Baseline
{
    /// <summary>
    /// SPEC §7: vanilla un-compressed representation. Rows joined with " | ",
    /// separated by "\n", no escaping, no address prefixes. Whatever a caller
    /// would otherwise paste into a prompt before applying SheetCompressor.
    /// </summary>
    public static string VanillaEncode(Grid grid)
    {
        return string.Join("\n", grid.Rows.Select(row => string.Join(" | ", row)));
    }
}

File: packages/go/sheetcompressor/baseline.go
package sheetcompressor

import "strings"

// vanillaEncode is SPEC §7's un-compressed baseline: rows joined with " | ",
// no escaping, no addresses. Used for rawBaseline.tokenEstimate.
func vanillaEncode(grid *Grid) string {
	lines := make([]string, len(grid.Rows))
	for i, row := range grid.Rows {
		lines[i] = strings.Join(row, " | ")
	}
	return strings.Join(lines, "\n")
}
